### PR TITLE
tbb: stop duplicating python bindings

### PR DESCRIPTION
**Pull request by branchv**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

Python bindings were being installed twice, one with `cmake` and another with `pip`. Dropped the `cmake` one since it installs a deprecated egg:

```console
$ pip3.12 list
DEPRECATION: Loading egg at /opt/homebrew/Cellar/tbb/2021.12.0/lib/python3.12/site-packages/TBB-0.2-py3.12-macosx-14.0-arm64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
```

### CODE REVIEW

**Comment by branchvincent** [comment]
On Linux: `/usr/bin/ld: cannot find -lirml: No such file or directory`

I guess we still need cmake for Linux then since it's also building a C++ library: https://github.com/oneapi-src/oneTBB/blob/224eca369a344bdd872066e66dbfd6289359546b/python/CMakeLists.txt#L57-L60

File: Formula/t/tbb.rb [hunk 1]
```diff
@@ -27,37 +26,26 @@ def python3
...
-
     args = %w[
       -DTBB_TEST=OFF
-      -DTBB4PY_BUILD=ON
+      -DTBB4PY_BUILD=OFF
     ]
 
     system "cmake", "-S", ".", "-B", "build/shared",
                     "-DBUILD_SHARED_LIBS=ON",
-                    "-DCMAKE_INSTALL_RPATH=#{rpath}",
```
Comment (carlocab) [inline Formula/t/tbb.rb:45]: I think you still need this

File: Formula/t/tbb.rb [hunk 2]
```diff
@@ -27,37 +26,26 @@ def python3
   end
 
   def install
-    # Prevent `setup.py` from installing tbb4py directly into HOMEBREW_PREFIX.
-    # We need this due to our Python patch.
-    site_packages = Language::Python.site_packages(python3)
-    inreplace "python/CMakeLists.txt", "install --prefix build -f", "\\0 --install-lib build/#{site_packages}"
-
-    tbb_site_packages = prefix/site_packages/"tbb"
-    ENV.append "LDFLAGS", "-Wl,-rpath,#{rpath},-rpath,#{rpath(source: tbb_site_packages)}"
```
Comment (carlocab) [inline Formula/t/tbb.rb:36]: And this too, but perhaps only for the Python build.

File: Formula/t/tbb.rb (hunk 1 above)
Comment (branchvincent) [reply Formula/t/tbb.rb]: oops, didn't mean to remove that one

File: Formula/t/tbb.rb (hunk 2 above)
Comment (branchvincent) [reply Formula/t/tbb.rb]: fixed

**Review by carlocab — APPROVED** [review]

**Pull request merged**